src/main/java/software/amazon/encryption/s3/internal/CipherSubscriber.java

```diff
@@ -91,21 +123,63 @@ public void onError(Throwable t) {
 
     @Override
     public void onComplete() {
+        // In rare cases, e.g. when the last part of a low-level MPU has 0 length,
+        // onComplete will be called before onNext is called once.
+        if (contentRead.get() + tagLength <= contentLength) {
+            finalBytes();
+        }
```

Comment:
Should this just be `<`?
```suggestion
        if (contentRead.get() + tagLength < contentLength) {
            finalBytes();
        }
```
Since if they're equal, then we know the finalBytes already happened? Maybe?
Either way, if finalBytes already happened, then `finalBytesCalled` would guard against this, so this is fine.